programs/local/LocalServer.cpp

```diff
@@ -105,6 +106,17 @@ void LocalServer::initialize(Poco::Util::Application & self)
         auto loaded_config = config_processor.loadConfig();
         config().add(loaded_config.configuration.duplicate(), PRIO_DEFAULT, false);
     }
+
+    GlobalThreadPool::initialize(
```

Comment:
We should check that the threads are created lazily.
Otherwise it will be excessive in most cases and also will make `gdb` and `strace` harder to look.

Reply:
The error was about IOThreadPool, and why do we need to also initialize GlobalThreadPool?

Reply:
`for i in {1..100}; do command time clickhouse-local --query "SELECT 1"; done`

This benchmark should not become worse.

Reply:
> The error was about IOThreadPool, and why do we need to also initialize GlobalThreadPool?

It uses the global thread pool.